source/MaterialXCore/Element.cpp

```diff
@@ -32,6 +32,9 @@ const string ValueElement::UI_NAME_ATTRIBUTE = "uiname";
 const string ValueElement::UI_FOLDER_ATTRIBUTE = "uifolder";
 const string ValueElement::UI_MIN_ATTRIBUTE = "uimin";
 const string ValueElement::UI_MAX_ATTRIBUTE = "uimax";
+const string ValueElement::UI_SOFT_MIN_ATTRIBUTE = "uisoftmin";
```

Comment:
@niklasharrysson . Another question for you. Is uimin/uimax hard min and max, or did we want the reverse logic that uimin, uimax are soft min/max and we need to add in a min/max?

Reply:
@bernardkwok I think the current logic is good. So basically we let the UI handle clamping of values, by having uimin/uimax represent hard limits for the UI, and let uisoftmin/uisoftmax be the initial UI widget ranges. So we avoid having to handle clamping in the core API.

Reply:
Sounds good.